codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -764,7 +764,9 @@ public boolean isServer() {
         @Override
         public DefaultStream reservePushStream(int streamId, Http2Stream parent) throws Http2Exception {
             if (parent == null) {
-                throw connectionError(PROTOCOL_ERROR, "Parent stream missing");
+                incrementExpectedStreamId(streamId);
+                throw streamError(streamId, Http2Error.REFUSED_STREAM, "PUSH_PROMISE sent on unknown stream, "
+                                                                       + "likely discarded due to a RST_STREAM.");
```

Comment:
I think this is better, but I'm not sure we're going to reach it: we're still using `shouldIgnoreHeadersOrDataFrame` in `DefaultHttp2ConnectionDecoder.onPushPromiseRead` which will return early if the parent stream is `null`.

Reply:
Yeah forgot to update the decoder as well. I'll push a fix today.

Reply:
@bryce-anderson can you take another look?